stubs/regex/regex/regex.pyi

```diff
@@ -0,0 +1,315 @@
+from typing import Any, AnyStr, Callable, overload
+
+import regex._regex as _regex
+from regex._regex_core import *
+
+__version__: str
+
+@overload
+def compile(pattern: AnyStr, flags: int = ..., ignore_unused: bool = ..., **kwargs: Any) -> _regex.Pattern[AnyStr]: ...
+@overload
+def compile(
+    pattern: _regex.Pattern[AnyStr], flags: int = ..., ignore_unused: bool = ..., **kwargs: Any
+) -> _regex.Pattern[AnyStr]: ...
```

Comment:
```suggestion
def compile(
    pattern: AnyStr |  _regex.Pattern[AnyStr], flags: int = ..., ignore_unused: bool = ..., **kwargs: Any
) -> _regex.Pattern[AnyStr]: ...
```

From what I can see, the following functions in `regex/regex/regex.pyi` don't really need overloads (they have the same return type in both overloads, so can be expressed more simply using a union-type for the `pattern` argument):

- `compile`
- `search`
- `match`
- `fullmatch`
- `split`
- `splititer`
- `findall`
- `finditer`
- `sub`
- `subf`
- `subn`
- `template`

Reply:
will do!

Reply:
Fixed these, I get how overrides work now thanks! 😄

Reply:
No problem!

Reply:
@AlexWaygood out of curiosity is there a reason why `@overload` is used over union-type for a similar situation in `re.pyi` given that they both versions also just return the same thing?

https://github.com/python/typeshed/blob/7fc402a6d05a48550d74ffd1b0e14db8afee787b/stdlib/re.pyi#L54-L57

Reply:
Those should probably be combined, but in a separate PR.